File: LinkedList.cpp
# include <stdio.h>
struct Node {
    int data;
    Node *next;
};
Node *root = NULL;

void rightPush(int data)
{
    if (root == NULL) {
        root = new Node();
        root->data = data;
        root->next = NULL;
    } else {
        Node *currentNode = root;
        while(currentNode->next != NULL) {
            currentNode = currentNode->next;
        }
        Node *newNode = new Node();
        newNode->data = data;
        newNode->next = NULL;
        currentNode->next = newNode;
    }
}

void leftPush(int data)
{
    if(root == NULL) {
        root = new Node();
        root->data = data;
        root->next = NULL;
    } else {
        Node *currentRoot = root;
        Node *newNode = new Node();
        newNode->data = data;
        newNode->next = currentRoot;
        root = newNode;
    }
}

void popLeft()
{
    if (root != NULL && root->next == NULL) {
        Node *temp = root;
        root = root->next;
        delete(temp);
    } else if(root != NULL && root->next != NULL) {
        Node *currentRoot = root;
        root = currentRoot->next;
        delete(currentRoot);
    }
}

void popRight()
{
    if(root != NULL && root->next == NULL) {
        Node *temp = root;
        root = root->next;
        delete(temp);
    } else if(root != NULL && root->next != NULL) {
        Node *currentNode = root;
        Node *previousNode = NULL;
        while(currentNode->next != NULL) {
            previousNode = currentNode;
            currentNode = currentNode->next;
        }
        delete(currentNode);
        previousNode->next = NULL;
    }
}

void pop(int data)
{
    if (root->data == data) {
        if(root->next == NULL) {
            Node *temp = root;
            root = root->next;
            delete(temp);
        } else {
            Node *currentNode = root;
            root = root->next;
            delete(currentNode);
        }
    } else if (root != NULL) {
        Node *currentNode = root;
        Node *previousNode = NULL;
        while(currentNode->data != data) {
            previousNode = currentNode;
            currentNode = currentNode->next;
        }
        previousNode->next = currentNode->next;
        delete(currentNode);
    }
}

void print()
{
    Node *currentNode = root;
    while(currentNode != NULL) {
        printf("%d ", currentNode->data);
        currentNode = currentNode->next;
    }
}

int main()
{
    rightPush(34);
    rightPush(2);
    rightPush(4);
    leftPush(23);
    pop(4);
    print();
    return 0;
}

File: LinkedList.py
class Node(object):
	def __init__(self, data):
		self.data = data
		self.next = None

class LinkedList(object):
	def __init__(self):
		self.root = None

	def rightPush(self, data):
		if self.root == None:
			self.root = Node(data)
		else:
			currentNode = self.root
			while currentNode.next is not None:
				currentNode = currentNode.next
			newNode = Node(data)
			currentNode.next = newNode

	def leftPush(self, data):
		if self.root == None:
			self.root = Node(data)
		else:
			currentNode = self.root
			newNode = Node(data)
			newNode.next = currentNode
			self.root= newNode

	def rightPop(self):
		if self.root and self.root.next == None:
			temp = self.root
			self.root = self.root.next
			del temp
		elif self.root:
			currentNode = self.root
			previousNode = None
			while currentNode.next is not None:
				previousNode = currentNode
				currentNode = currentNode.next
			del currentNode
			previousNode.next = None

	def leftPop(self):
		if self.root and self.root.next == None:
			temp = self.root
			self.root = self.root.next
			del temp
		elif self.root:
			currentNode = self.root
			self.root = currentNode.next
			del currentNode

	def pop(self, data):
	 	if self.root and self.root.data == data:
	 		self.root = self.root.next
	 	elif self.root:
	 		currentNode = self.root
	 		previousNode = None
	 		while currentNode.data is not data:
	 			previousNode = currentNode
	 			currentNode = currentNode.next
	 		previousNode.next = currentNode.next
	 		del currentNode


	def show(self):
		currentNode = self.root
		while currentNode is not None:
			print(currentNode.data)
			currentNode = currentNode.next

s = LinkedList()
s.rightPush(1)
s.rightPush(2)
s.rightPush(10)
s.rightPush(3)
s.rightPush(3)
s.pop(10)
s.show()

